webapp/src/templates/partials/reports_content.html

```diff
@@ -68,7 +68,16 @@
           </li>
           <li ng-if="selection.formatted.form && selection.formatted.content_type === 'xml'" ng-repeat="field in selection.displayFields" class="indent-{{field.depth}}">
             <label translate>{{field.label}}</label>
-            <p ng-hide="field.imagePath">{{field.value}}</p>
+            <p ng-hide="field.imagePath">
+              <span ng-if="labelIsIDorName(field.label)">
+                  <a class="name" ng-if="summary.subject._id || summary.fields.patient_uuid" ui-sref="contacts.detail({ id: summary.subject._id || summary.fields.patient_uuid })"
+                    ng-bind="field.value">
+                  </a>
+              </span>
+              <span ng-if="!labelIsIDorName(field.label)">
+                  {{ field.value }}
```

Comment:
Indentation is off here.

Reply:
Just noticed the unchanged block you highlighted above i.e. `selection.formatted.content_type !== 'xml'` has no `translation` attribute  on the `label` html field, our function `labelIsIDorName` will not work with it, how do we go about it?

Reply:
Right, that will need some investigation. I think the FormatDataRecord might do the translation. FormatDataRecord is an ancient block of code (it's been at Medic longer than I have!). There are two approaches I can think of - a) move the translation out of the service and in to the template, or b) update the service so the formatted record has a URL property if it should link somewhere.

I like the second option because it would work for both xml and json reports, and it's also relatively easy to unit test. If you want to go with option b you can use the `href` function on the `$state` service to generate the URL in code - [documentation](https://ui-router.github.io/ng1/docs/0.3.1/index.html#/api/ui.router.state.$state)

Reply:
Well, sounds like I'll need more time to wrap my head around this, can we spin it off from this ticket?

Reply:
The [original ticket](https://github.com/medic/medic-projects/issues/4458) was about an SMS report so I think it's critical to solving this issue.